addons/portal/static/src/js/portal_chatter.js

```diff
@@ -43,8 +62,98 @@ var PortalComposer = publicWidget.Widget.extend({
     /**
      * @private
      */
+    _onAttachmentButtonClick: function () {
+        this.$fileInput.click();
+    },
+    /**
+     * @private
+     * @param {Event} ev
+     * @returns {Promise}
+     */
+    _onAttachmentDeleteClick: function (ev) {
+        var self = this;
+        var attachmentId = $(ev.currentTarget).closest('.o_portal_chatter_attachment').data('id');
+        var accessToken = _.find(this.attachments, {'id': attachmentId}).access_token;
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        this.$sendButton.prop('disabled', true);
+
+        return this._rpc({
+            route: '/portal/attachment/remove',
+            params: {
+                'attachment_id': attachmentId,
+                'access_token': accessToken,
+            },
+        }).then(function () {
+            self.attachments = _.reject(self.attachments, {'id': attachmentId});
+            self._updateAttachments();
+            self.$sendButton.prop('disabled', false);
+        });
+    },
+    /**
+     * @private
+     * @returns {Promise}
+     */
+    _onFileInputChange: function () {
+        var self = this;
+        var uploadMutex = new concurrency.Mutex();
+
+        this.$sendButton.prop('disabled', true);
+
+        _.each(this.$fileInput[0].files, function (file) {
+            // Upload one file at a time: no need to parallel as upload is
+            // limited by bandwidth.
+            uploadMutex.exec(function () {
```

Comment:
I think there is an ideal number (around 5 or 6?) where it is better to parallel, but after which it becomes counter productive.

For the sake of code simplicity, and because the use case here is not to upload 100s of files, I'll remove the mutex.